src/k8s/pkg/component/component.go

```diff
@@ -44,52 +54,68 @@ type Component struct {
 	Status bool
 }
 
+// InitializeHelmClientConfig initializes a Helm Configuration, ensures the use of a fresh configuration
+func (r *HelmClientIntitializer) InitializeHelmClientConfig() (*action.Configuration, error) {
+	settings := cli.New()
+	settings.KubeConfig = "/etc/kubernetes/admin.conf"
+
+	actionConfig := new(action.Configuration)
+	err := actionConfig.Init(
+		settings.RESTClientGetter(),
+		settings.Namespace(),
+		os.Getenv("HELM_DRIVER"),
+		logAdapter,
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize action config: %w", err)
+	}
+	return actionConfig, nil
+}
+
 func logAdapter(format string, v ...any) {
 	logrus.Debugf(format, v...)
 }
 
 // NewManager creates a new Component manager instance.
-func NewManager(snap snap.Snap) (*helmClient, error) {
+func NewManager(snap snap.Snap, initializers ...HelmConfigInitializer) (*helmClient, error) {
+	var initializer HelmConfigInitializer
+
+	if len(initializers) > 0 {
+		initializer = initializers[0]
+	} else {
+		// If no initializer provided, use a default one
+		initializer = &HelmClientIntitializer{}
+	}
```

Comment:
Is this to allow `NewManager(snap)` to just work and not having to update callers? I'd much refer to be explicit here, and update callers as needed. Let's also make this `NewHelmManager`, to indicate that this uses helm underneath and not a generic implementation (the arguments already are helm specific)

```suggestion
func NewHelmManager(snap snap.Snap, initializer HelmConfigInitializer) (*helmClient, error) {
	if initializer == nil {
		initializer = &HelmClientIntitializer{}
	}
```

Reply:
This also needs updating all callers, and they should do `NewHelmManager(snap, nil)`

Reply:
^ this is not addressed, @louiseschmidtgen

Reply:
You are too quick, I was just updating it :)